components/core/services/resize.ts

```diff
@@ -31,6 +31,12 @@ export class NzResizeService {
     this.renderer = this.rendererFactory2.createRenderer(null, null);
   }
 
+  ngOnDestroy(): void {
+    // Caretaker note: the `handler` is an instance property (it's not defined on the class prototype).
+    // The `handler` captures `this` and prevents the `NzResizeService` from being GC'd.
+    this.handler = NOOP;
```

Comment:
IMHO, the memory is freed when the `NzResizeService` destroys. Could you please make a reproduction indicating that it do leak here?